arrows/core/video_input_image_list.cxx

```diff
@@ -23,182 +23,191 @@
 
+// ----------------------------------------------------------------------------
 class video_input_image_list::priv
 {
 public:
   priv( video_input_image_list* parent )
-  : m_parent( parent )
-  , m_current_file( m_files.end() )
-  , m_frame_number( 0 )
-  , m_image( nullptr )
-  , m_have_metadata_map( false )
+    : m_parent{ parent },
+      m_current_file{ m_files.end() }
   {}
 
-  video_input_image_list* m_parent;
+  video_input_image_list* const m_parent;
 
   // Configuration values
   std::vector< std::string > c_search_path;
   std::vector< std::string > c_allowed_extensions;
 
-  // local state
-  std::vector < kwiver::vital::path_t > m_files;
-  std::vector < kwiver::vital::path_t >::const_iterator m_current_file;
-  kwiver::vital::frame_id_t m_frame_number;
-  kwiver::vital::image_container_sptr m_image;
+  // Local state
+  std::vector< kv::path_t > m_files;
+  std::vector< kv::path_t >::const_iterator m_current_file;
+  kv::frame_id_t m_frame_number = 0;
+  kv::image_container_sptr m_image;
 
-  // metadata map
-  bool m_have_metadata_map;
+  // Metadata map
+  bool m_have_metadata_map = false;
   vital::metadata_map::map_metadata_t m_metadata_map;
-  std::map< kwiver::vital::path_t, kwiver::vital::metadata_sptr > m_metadata_by_path;
+  std::map< kv::path_t, kv::metadata_sptr > m_metadata_by_path;
 
-  // processing classes
+  // Processing classes
   vital::algo::image_io_sptr m_image_reader;
 
   void read_from_file( std::string const& filename );
   void read_from_directory( std::string const& dirname );
   vital::metadata_sptr frame_metadata(
-    const kwiver::vital::path_t& file,
-    kwiver::vital::image_container_sptr image = nullptr );
+    kv::path_t const& file, kv::image_container_sptr image = nullptr );
 };
 
-
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
 video_input_image_list
 ::video_input_image_list()
   : d( new video_input_image_list::priv( this ) )
 {
   attach_logger( "arrows.core.video_input_image_list" );
 
-  set_capability( vital::algo::video_input::HAS_EOV, true );
-  set_capability( vital::algo::video_input::HAS_FRAME_NUMBERS, true );
-  set_capability( vital::algo::video_input::HAS_FRAME_DATA, true );
-  set_capability( vital::algo::video_input::HAS_METADATA, true );
+  set_capability( HAS_EOV, true );
+  set_capability( HAS_FRAME_NUMBERS, true );
+  set_capability( HAS_FRAME_DATA, true );
+  set_capability( HAS_METADATA, true );
 
-  set_capability( vital::algo::video_input::HAS_FRAME_TIME, false );
-  set_capability( vital::algo::video_input::HAS_ABSOLUTE_FRAME_TIME, false );
-  set_capability( vital::algo::video_input::HAS_TIMEOUT, false );
-  set_capability( vital::algo::video_input::IS_SEEKABLE, true );
+  set_capability( HAS_FRAME_TIME, false );
+  set_capability( HAS_ABSOLUTE_FRAME_TIME, false );
+  set_capability( HAS_TIMEOUT, false );
+  set_capability( IS_SEEKABLE, true );
```

Comment:
Why are the namespaces removed here? In other places I see vital::algo shortened to kv::algo, what is the rational of not doing that here? 

My personal preference is when I can statically glance at at variable and understand where its definition is. That style seems to be used everywhere else in this file, why not here?

Reply:
...because they are in scope and therefore superfluous. Are there other instances where superfluous qualification is used? (I don't recall any.)

OTOH if there is a strong opinion that they should be qualified, even though they don't need to be, I suppose I can put them back...

Reply:
IMO it makes it a lot easier to understand where they come from. It reduces the search space from any of the headers in the include tree to something that is working in the `kv::algo::video_input` (and it gives a strong hint as to which of those files it is). But this is C++ and not Python, so I don't want to impose unnecessary style constraints based on another language's sensibilities. However, IMO, I do think it improves readability. Those are just my 2cents.

Reply:
I don't care too strongly one way or the other (as long as we are consistent).  However, I will say that since C++ is strongly typed, it is less ambiguous about where those enums came from.  The only values you can put in there (without type casting) are those defined in the enum used by this function.  If you can find the `set_capabilities` function definition then you can easily find the enum.  I guess I'm slightly in favor of not using the namespaces.

Reply:
...except they *aren't* enums. They're static members that are type-aliases of `std::string`.

Anyway, I split the difference (also w.r.t. prior use of `super`) by hoisting `video_input` into scope, similar to how I was already doing with `image_io`, and qualified them with that.